napari/experimental/_progressive_loading.py

```diff
@@ -0,0 +1,1652 @@
+    if np.any((bottom_right - top_left) == 0):
+        return
+
+    # TODO we could add padding around top_left and bottom_right to account
+    #      for future camera movement
+
+    # Interval must be nonnegative
+    if not np.all(top_left <= bottom_right):
+        import pdb
+
+        pdb.set_trace()
+
+    corners = viewer.layers[get_layer_name_for_scale(0)].corner_pixels
+
+    LOGGER.info(
+        f"dims_update_handler: start render_sequence {corners} on layer \
+        {get_layer_name_for_scale(0)}"
+    )
+
+    # Find the visible scales
+    visible_scales = [False] * len(data.arrays)
+    min_scale = 0
+    max_scale = len(data.arrays) - 1
+
+    ndisplay = ndisplay if ndisplay else viewer.dims.ndisplay
+
+    # Get the scale visibility predicate for the correct ndisplay
+    should_render_scale = (
+        should_render_scale_2D if ndisplay == 2 else should_render_scale_3D
+    )
+
+    for scale in range(len(data.arrays)):
+        layer_name = get_layer_name_for_scale(scale)
+        layer = viewer.layers[layer_name]
+
+        layer.metadata["translated"] = False
+
+        # Reenable visibility of layer
+        visible_scales[scale] = should_render_scale(
+            scale, viewer, min_scale, max_scale
+        )
+        layer.visible = visible_scales[scale]
+        layer.opacity = 0.9
+
+        LOGGER.info(
+            f"scale {scale} name {layer_name}\twith translate \
+            {layer.data.translate}"
+        )
+
+    # Update the MultiScaleVirtualData memory backing
+    data.set_interval(top_left, bottom_right, visible_scales=visible_scales)
+
+    # Start a new multiscale render
+    worker = render_sequence(
+        corners,
+        data=data,
+        visible_scales=visible_scales,
+        ndisplay=ndisplay,
+        camera=camera,
+    )
+
+    LOGGER.info(
+        f"dims_update_handler: started render_sequence with corners {corners}"
+    )
+
+    # This will consume our chunks and update the numpy "canvas" and refresh
+    def on_yield(coord):
+        # TODO bad layer access
+        chunk_slice, scale, scale_factors, chunk, is_last_chunk = coord
+
+        layer_name = get_layer_name_for_scale(scale)
+        layer = viewer.layers[layer_name]
+
+        # TODO this relies on the coincidence that node indices are 2 or 3 for
+        #      the image and volume members of an Image layer
+        node = viewer.window.qt_viewer.layer_to_visual[
+            layer
+        ]._layer_node.get_node(viewer.dims.ndisplay)
+
+        texture = node._texture
+
+        LOGGER.info(
+            f"Writing chunk with size {chunk.shape} to: \
+            {(scale, [(sl.start, sl.stop) for sl in chunk_slice])} in layer \
+            {scale} with shape {layer.data.shape} and dataplane shape \
+            {layer.data.hyperslice.shape} sum {chunk.sum()}"
+        )
+
+        if not layer.metadata["translated"]:
+            layer.translate = (
```

Comment:
We currently have `Layer._transforms['tile2data']` that is used for something similar to this. Don't think you need to make any changes in this PR to handle that, but should be interesting for future/long-term discussions.